src/mpl2/src/hier_rtlmp.cpp

```diff
@@ -704,33 +708,39 @@ Metrics* HierRTLMP::computeMetrics(odb::dbModule* module)
 
   Metrics* metrics
       = new Metrics(num_std_cell, num_macro, std_cell_area, macro_area);
+
   logical_module_map_[module] = metrics;
+
   return metrics;
 }
 
-// compute the metrics for a cluster
-// Here we do not include any Pads,  Covers or Marker
-// number of standard cells
-// number of macros
-// area of standard cells
-// area of macros
+float HierRTLMP::computeMicronArea(odb::dbInst* inst)
+{
+  const float width = dbuToMicron(inst->getBBox()->getBox().dx(), dbu_);
+  const float height = dbuToMicron(inst->getBBox()->getBox().dy(), dbu_);
+
+  return width * height;
+}
+
 void HierRTLMP::setClusterMetrics(Cluster* cluster)
 {
-  unsigned int num_std_cell = 0;
-  unsigned int num_macro = 0;
-  float std_cell_area = 0.0;
-  float macro_area = 0.0;
-  num_std_cell += cluster->getLeafStdCells().size();
-  num_macro += cluster->getLeafMacros().size();
-  for (odb::dbInst* inst : cluster->getLeafStdCells()) {
-    const sta::LibertyCell* liberty_cell = network_->libertyCell(inst);
-    std_cell_area += liberty_cell->area();
```

Comment:
@maliberty @AcKoucher When I fix a bug, I try to remember to ask myself if this bug exists elsewhere too: there are 25 other places `->area()` is used in OpenROAD, should any of the other be reconsidered?

Reply:
I considered it.  Note that not all ->area() are calls in sta as there are other methods with the same name that do not rely on sta.

The problem shows up in uW as the rams were generated a long time ago when we didn't write the area into Liberty abstracts.  That was fixed a long time ago but we don't have a good way to regenerate these.  It is isn't a problem on std cell libraries where most of these calls are relevant.

There are two uses in par that are worth looking at @AcKoucher

Reply:
If the only issue is the area property on those .libs, can't they just be added manually?

Reply:
How about a warning when .lib is broken, delete the broken fakrams and replace them with mock srams that are more easily maintained?